lib/dartdoc.dart

```diff
@@ -78,108 +65,98 @@ Future<List<Generator>> initGenerators(String url, String relCanonicalPrefix,
-  final List<String> includes;
-  final List<String> includeExternals;
-  final List<String> excludes;
   final Set<String> writtenFiles = new Set();
 
   // Fires when the self checks make progress.
   final StreamController<String> _onCheckProgress =
       new StreamController(sync: true);
 
-  Stopwatch _stopwatch;
-
-  DartDoc(this.rootDir, this.excludes, this.sdkDir, this.generators,
-      this.outputDir, this.packageMeta, this.includes,
-      {this.includeExternals: const []});
+  DartDoc(
+      Directory rootDir,
+      List<String> excludes,
+      Directory sdkDir,
+      this.generators,
+      this.outputDir,
+      PackageMeta packageMeta,
+      List<String> includes,
+      List<String> includeExternals)
+      : super(
+            rootDir,
+            excludes,
+            config.excludePackages,
+            sdkDir,
+            packageMeta,
+            includes,
+            includeExternals,
+            config.showWarnings,
+            config.autoIncludeDependencies);
 
   Stream<String> get onCheckProgress => _onCheckProgress.stream;
 
-  /// Generate DartDoc documentation.
-  ///
-  /// [DartDocResults] is returned if dartdoc succeeds. [DartDocFailure] is
-  /// thrown if dartdoc fails in an expected way, for example if there is an
-  /// analysis error in the code. Any other exception can be throw if there is an
-  /// unexpected failure.
-  Future<DartDocResults> generateDocs() async {
-    _stopwatch = new Stopwatch()..start();
-
-    List<String> files = packageMeta.isSdk
-        ? const []
-        : findFilesToDocumentInPackage(rootDir.path).toList();
-
-    // TODO(jcollins-g): seems like most of this belongs in the Package constructor
-    List<LibraryElement> libraries = _parseLibraries(files, includeExternals);
-
-    if (includes != null && includes.isNotEmpty) {
-      Iterable knownLibraryNames = libraries.map((l) => l.name);
-      Set notFound =
-          new Set.from(includes).difference(new Set.from(knownLibraryNames));
-      if (notFound.isNotEmpty) {
-        throw 'Did not find: [${notFound.join(', ')}] in '
-            'known libraries: [${knownLibraryNames.join(', ')}]';
+  @override
+  void logAnalysisErrors(Set<Source> sources) {
+    List<AnalysisErrorInfo> errorInfos = [];
+    // TODO(jcollins-g): figure out why sources can't contain includeExternals
+    // or embedded SDK components without having spurious(?) analysis errors.
+    // That seems wrong.
+    for (Source source in sources) {
+      context.computeErrors(source);
+      AnalysisErrorInfo info = context.getErrors(source);
+      List<_Error> errors = [info]
+          .expand((AnalysisErrorInfo info) {
+            return info.errors.map((error) =>
+                new _Error(error, info.lineInfo, packageMeta.dir.path));
+          })
+          .where((_Error error) => error.isError)
+          .toList()
+            ..sort();
+      // TODO(jcollins-g): Why does the SDK have analysis errors?  Annotations
```

Comment:
Perhaps also track as an issue?

Reply:
done, and issue added to relevant TODOs.